### ISSUE TITLE
PDF is taking time so want to show loading icon

### ISSUE THREAD

**Issue by jpsiet** [eKoopmans/html2pdf#80]
PDF is taking time so want to show loading icon. Can you please let me know can i do.. like any event trigger ...

**Comment by eKoopmans** [comment]
Hey @jpsiet, I have just pushed a new branch, [new-api](https://github.com/eKoopmans/html2pdf/tree/new-api), which adds Promises! Using it, you could easily run a function when html2pdf is finished, like so:

```js
// Create the PDF, prompt the user to save, and then call your function:
html2pdf(element, opt).then(function () {
  console.log('done!');
});
```

[The readme](https://github.com/eKoopmans/html2pdf/tree/new-api#advanced-usage) has all the information on the new API. Please let me know if there are any problems with it - I want to make sure it's stable before merging into master!

**Comment by eKoopmans** [comment]
Hi, closing as it's resolved by the v0.9.0 new API.
